base/subarray2.jl

```diff
@@ -1,107 +1,47 @@
 ## Scalar indexing
-# Low dimensions: avoid splatting
-newsym = (:i_1, :i_2, :i_3, :i_4)
-vars = Array(Expr, 0)
-varsInt = Array(Expr, 0)
-varsOther = Array(Expr, 0)
-vars_toindex = Array(Expr, 0)
-for i = 1:4
-    sym = newsym[i]
-    push!(vars, Expr(:quote, sym))
-    push!(varsInt, :($sym::Int))
-    push!(varsOther, :($sym::Union(Real, AbstractVector)))
-    push!(vars_toindex, :(to_index($sym)))
-    ex = i == 1 ? quote
-         getindex{T,N,P,IV}(V::SubArray{T,N,P,IV}, $sym::Real) = getindex(V, to_index($sym))
-        setindex!{T,N,P,IV}(V::SubArray{T,N,P,IV}, v, $sym::Real) = setindex!(V, v, to_index($sym))
-         getindex{T,N,P,IV}(V::SubArray{T,N,P,IV}, $sym::AbstractVector{Bool}) = getindex(V, to_index($sym))
-        setindex!{T,N,P,IV}(V::SubArray{T,N,P,IV}, v, $sym::AbstractVector{Bool}) = setindex!(V, v, to_index($sym))
-    end : quote
-         getindex{T,N,P,IV}(V::SubArray{T,N,P,IV}, $(varsOther...)) = getindex(V, $(vars_toindex...))
-        setindex!{T,N,P,IV}(V::SubArray{T,N,P,IV}, v, $(varsOther...)) = setindex!(V, v, $(vars_toindex...))
-    end
-    @eval begin
-        @generated function getindex{T,N,P,IV,LD}(V::SubArray{T,N,P,IV,LD}, $(varsInt...))
-            if $i == 1 && length(IV.parameters) == LD  # linear indexing
-                meta = Expr(:meta, :inline)
-                if iscontiguous(V)
-                    return :($meta; V.parent[V.first_index + i_1 - 1])
-                end
-                return :($meta; V.parent[V.first_index + V.stride1*(i_1-1)])
-            end
-            exhead, ex = index_generate(ndims(P), IV, :V, [$(vars...)])
-            quote
-                $exhead
-                $ex
-            end
+@inline getindex(V::SubArray, I::Int...) = (checkbounds(V, I...); unsafe_getindex(V, I...))
+@generated function unsafe_getindex{T,N,P,IV,LD}(V::SubArray{T,N,P,IV,LD}, I::Int...)
+    ni = length(I)
+    if ni == 1 && length(IV.parameters) == LD  # linear indexing
+        meta = Expr(:meta, :inline)
+        if iscontiguous(V)
+            return :($meta; V.parent[V.first_index + I[1] - 1])
```

Comment:
Missed this comment: yes, this is largely the same as the existing code, just not indented within an `@eval` block anymore.